build/Settings.cs

```diff
@@ -56,5 +57,76 @@ private static string config(string @default = null, [CallerMemberName] string k
         public static string CommitId => config(null, "Build.SourceVersion") ?? config("N/A", "APPVEYOR_REPO_COMMIT");
 
         public static string BuildArtifactsStorage => config(null);
+
+        public class SignInfo
+        {
+            public static string AzureSigningConnectionString => config(null, "AzureBlobSigningConnectionString");
+            public static readonly string AzureToSignContainerName = "azure-functions-cli";
+            public static readonly string AzureSignedContainerName = "azure-functions-cli-signed";
+            public static readonly string AzureSigningJobName = "signing-jobs";
+            public static readonly string ToSignZipName = $"{BuildNumber}-authenticode.zip";
+            public static readonly string ToSignThirdPartyName = $"{BuildNumber}-third.zip";
+            public static readonly string ToSignDir = "unsigned";
+            public static readonly string SignedDir = "signed";
+            public static readonly string Authenticode = "SignAuthenticode";
+            public static readonly string ThirdParty = "Sign3rdParty";
+            public static readonly string[] RuntimesToSign = new[] {"min.win-x86"};
+            public static readonly string[] filterExtenstionsSign = new[] { ".json", ".spec", ".cfg", ".pdb", ".config", ".nupkg", ".py" };
+
+            public static readonly string[] authentiCodeBinaries = new[] {
+                "DurableTask.AzureStorage.Internal.dll",
+                "DurableTask.Core.Internal.dll",
+                "func.deps.json",
+                "func.dll",
+                "func.dll.config",
+                "func.exe",
+                "func.pdb",
+                "func.runtimeconfig.json",
+                "Microsoft.Azure.AppService.Proxy.Client.dll",
+                "Microsoft.Azure.AppService.Proxy.Common.dll",
+                "Microsoft.Azure.AppService.Proxy.Runtime.dll",
+                "Microsoft.Azure.WebJobs.dll",
+                "Microsoft.Azure.WebJobs.Extensions.Http.dll",
+                "Microsoft.Azure.WebJobs.Host.dll",
+                "Microsoft.Azure.WebJobs.Host.Storage.dll",
+                "Microsoft.Azure.WebJobs.Logging.ApplicationInsights.dll",
+                "Microsoft.Azure.WebJobs.Logging.dll",
+                "Microsoft.Azure.WebJobs.Script.dll",
+                "Microsoft.Azure.WebJobs.Script.Grpc.dll",
+                "Microsoft.Azure.WebJobs.Script.WebHost.dll",
+                "Microsoft.Azure.WebSites.DataProtection.dll",
+                Path.Combine("templates", "itemTemplates.2.0.10328.nupkg"),
```

Comment:
@mhoeger  is the signing process able to sign nupkgs in the zip file directly?

Reply:
Depends on the signing... for host, we strong name sign all of the .dll's in the nupkg and then nuget sign the entire package. I don't know how Authenticode signing works with nupkgs. @ahmelsayed, i think you or fabio would know more on that! 

So to answer your question: not sure if nupkgs are able to be authenticode signed, but we do have a process for nuget signing and/or strong name signing the .dll's

Reply:
@ahmelsayed, I made these lists of the unsigned things and then filtered out things that we weren't able to sign from the scripts. Just so we have the lists properly and we are taking into account all the files.

Reply:
oh I see, sorry I missed the filtering below.

Reply:
For clarity, only the contents need to be authenticode signed. The nupkg should be signed like the others.